src/main/java/imsam/mgen/MultiTargetModelGenerator.java

```diff
@@ -0,0 +1,334 @@
+                    stateSpace[predecessor].transitionsOut.add(predTransition);
+                    stateSpace[stateId].transitionsIn.add(predTransition);
+                } else {
+                    logger.trace(predecessor+" already connects to "+stateId);
+                }
+            }
+        } else {
+            logger.debug("-------------All states are already on a target path-------------");
+        }
+
+        /*
+         * Once every state is on a target path, transitions are randomly generated to connect the model for every state except
+         * the initial state and target states. The successor state is not allowed to be the current state, and if the 
+         * transition already exists with a different rate then that transition is skipped. If it is a simple model 
+         * then this section is skipped.
+         */
+        if(!simple){
+            logger.debug("-------------Randomly generating other transitions-------------");
+            for(State state : stateSpace){
+                int numberOfTransitions =(int)transitionCountDistribution.random();
+                logger.trace("Generating "+ numberOfTransitions+ " transitions for "+ state.stateId);
+                if(!targets.contains(state.stateId) || state.stateId != 0 || state.stateId == absorbingState){
+                    for(int transitionId = 0; transitionId<numberOfTransitions; transitionId++){
+                        int successor = (int)(Math.random()*numberOfStates);
+                        int rate = (int)transitionRateDistribution.random(); 
+                        TransitionPath transition = new TransitionPath(
+                            state.stateId, 
+                            successor, 
+                            rate);
+                        if((!state.transitionOutExists(transition)) && (successor != state.stateId)){
+                            logger.trace("Connecting state " +state.stateId+ " to state "+ successor);
+                            state.transitionsOut.add(transition);
+                            stateSpace[successor].transitionsIn.add(transition);
+                        } else {
+                            logger.trace(state.stateId+" already connects to "+successor);;
+                        }
+                    }
+                } else {
+                    logger.trace("Skipping target state "+ state.stateId);
+                }
+            }
+        }
+        //Calculate runtime
+        long end = System.nanoTime();
+        logger.info(numberOfStates+"-state model generated in "+(end-start)/1000000+ "ms");
+    }
+
+    /**
+     * Randomly choose target states from state space, arbitrarily bounded at 1/3 of state space
+     * @return List of target states
+     */
+    private List<Integer> generateTargetStates(){
+        StringBuilder strBld = new StringBuilder();
+        List<Integer> targets = new ArrayList<>();
+        double third = 1/3.0f;
```

Comment:
This is only used once, so it would be bettet not to declare it as a variable. Using a double will not affect the result of calulatations since you conver back to int anyway.

Reply:
I removed the variable and now just divide the int by 3.